ark/utils/plot_utils.py

```diff
@@ -11,103 +11,100 @@
 from skimage.exposure import rescale_intensity
 
 
-# plotting functions
+def tif_overlay_preprocess(segmentation_labels, plotting_tif):
+    """Validates plotting_tif and preprocesses it accordingly
 
-def plot_overlay(predicted_contour, plotting_tif, alternate_contour=None, path=None, show=False):
+    Args:
+        segmentation_labels (numpy.ndarray):
+            2D numpy array of labeled cell objects
+        plotting_tif (numpy.ndarray):
+            2D or 3D numpy array of imaging signal
+
+    Returns:
+        numpy.ndarray:
+            The preprocessed image
+    """
+
+    if len(plotting_tif.shape) == 2:
+        if plotting_tif.shape != segmentation_labels.shape:
+            raise ValueError("plotting_tif and segmentation_labels array dimensions not equal.")
+        else:
+            # convert RGB image with same data across all three channels
+            formatted_tif = np.stack((plotting_tif, plotting_tif, plotting_tif), axis=2)
```

Comment:
this looks like the same logic as before for assigning channels a color. See my previous comment for the desired mapping between number of channels and which colors they should get.